src/test/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMockTest.java

```diff
@@ -411,4 +413,35 @@ void runTaskTimer_ZeroDelay_DoesntExecuteTaskImmediately_Consumer()
 		assertEquals(1, count.get());
 	}
 
+	@Test
+	void getMainThreadExecutor_RunsOnMainThread()
+	{
+		MockBukkit.mock();
+		Executor executor = scheduler.getMainThreadExecutor(MockBukkit.createMockPlugin());
+		AtomicBoolean b = new AtomicBoolean();
+
+		executor.execute(() -> b.set(true));
+		scheduler.performOneTick();
+
+		assertTrue(b.get());
+		MockBukkit.unmock();
+	}
```

Comment:
I think a much better way to test this is to use `b.set(Bukkit.isPrimaryThread())`.